src/features/pass-survey/lib/markdownVariableReplacer.ts

```diff
@@ -64,11 +71,78 @@ export class MarkdownVariableReplacer {
     return format(new Date(year, month, day), 'y-MM-dd');
   };
 
+  private parseBasicSystemVariables = (markdown: string) => {
+    return markdown
+      .replaceAll(/\[Now\]/gi, format(this.now, 'h:mm aa') + ' today (now)')
+      .replaceAll(/\[Nickname\]/gi, this.nickName)
+      .replaceAll(/\[sys.date\]/gi, format(this.now, 'MM/dd/y'));
+  };
+
+  private cleanUpUnusedResponseVariables = (markdown: string) => {
+    return markdown
+      .replaceAll(/\[Time_Elapsed_Activity_Last_Completed\]/gi, '')
+      .replaceAll(/\[Time_Activity_Last_Completed\]/gi, '');
+  };
+
+  public parseSystemVariables = (markdown: string) => {
+    if (!this.lastResponseTime) {
+      const cleanedUpMarkdown = this.cleanUpUnusedResponseVariables(markdown);
+      return this.parseBasicSystemVariables(cleanedUpMarkdown);
+    }
+
+    const markdownSplit = markdown?.split('\n');
+    markdownSplit.forEach((element, index) => {
+      if (element?.includes('[Time_Activity_Last_Completed] to [Now]')) {
+        markdownSplit[index] = `[blue]${markdownSplit[index]}`;
+        markdown = markdownSplit.join('\n');
+      }
+    });
+
+    return this.parseBasicSystemVariables(markdownSplit.join('\n'))
+      .replaceAll(
+        /\[Time_Elapsed_Activity_Last_Completed\]/gi,
+        this.formatTimeElapsed(),
+      )
+      .replaceAll(
+        /\[Time_Activity_Last_Completed\]/gi,
+        this.formatLastResponseTime(),
+      );
+  };
+
+  private formatTimeElapsed = () => {
```

Comment:
Although the function does some formatting stuff inside, the main purpose of it is to return a string.
So please consider another naming, like getTimeElapsed.
Format functions imply that they accept a string as an input (parameter) and return a string in a desired format as an output.